contracts/bonding/BondingManager.sol

```diff
@@ -860,21 +833,28 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
         totalBonded = totalBonded.add(_rewards);
     }
 
-    /*
-     * @dev Update a delegator with token pools shares from its lastClaimTokenPoolsSharesRound through a given round
+    /**
+     * @dev Update a delegator with token pools shares from its lastClaimRound through a given round
      * @param _delegator Delegator address
      * @param _endRound The last round for which to update a delegator's stake with token pools shares
      */
-    function updateDelegatorWithTokenPoolsShares(address _delegator, uint256 _endRound) internal {
+    function updateDelegatorWithEarnings(address _delegator, uint256 _endRound) internal {
         Delegator storage del = delegators[_delegator];
 
         // Only will have earnings to claim if you have a delegate
         // If not delegated, skip the earnings claim process
         if (del.delegateAddress != address(0)) {
+            // Cannot claim earnings for more than 20 rounds
+            // This is an arbitrary number to cause transactions to fail early if
+            // we know they will require too much gas to loop through all the necessary rounds to claim earnings
+            // The user should instead manually invoke `claimEarnings` to split up the claiming process
+            // across multiple transactions
+            require(_endRound.sub(del.lastClaimRound) <= 20);
```

Comment:
is 20 arbitrary or somewhat empirically observed as a reasonable value?

Reply:
Its arbitrary right now, but I can do some gas testing to see what a reasonable number of rounds to claim through is...